pkg/gofr/datasource/dbresolver/circuit_breaker.go

```diff
@@ -0,0 +1,55 @@
+package dbresolver
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+// Circuit breaker for replica health with atomic operations.
+type circuitBreaker struct {
+	failures    atomic.Int32
+	lastFailure atomic.Int64
+	state       atomic.Int32 // 0=closed, 1=open, 2=half-open.
```

Comment:
Here also, I might use atomic.Value

You could also use atomic.Pointer[circuitBreakerState]

By defining circuitBreakerState as the toe of all circuitBreakerState

The pointer is overkill here, but it makes it clear what you are dealing with